pages/PhentermineForm.tsx

```diff
@@ -0,0 +1,225 @@
+import { auth, submitPhentermineForm } from '../firebase/firebase'
+import router from 'next/router'
+
+function PhentermineForm() {
+  // State declarations for form fields
+  const [patientName, setPatientName] = useState('')
+  const [dateOfBirth, setDateOfBirth] = useState({
+    month: '',
+    day: '',
+    year: '',
+  })
+  const [patientSignature, setPatientSignature] = useState('')
+  const [witnessSignature, setWitnessSignature] = useState('')
+
+  // Sets the current date as default for the 'date' field
+  const [date, setDate] = useState(() => {
+    const today = new Date()
+    const formattedDate = `${today.getFullYear()}-${String(
+      today.getMonth() + 1,
+    ).padStart(2, '0')}-${String(today.getDate()).padStart(2, '0')}`
+    return formattedDate
+  })
+
+  const [initials, setInitials] = useState<string[]>(new Array(12).fill(''))
+
+  // Redirects to login page if no user is authenticated
+  useEffect(() => {
+    if (!auth.currentUser?.email) {
+      router.push('/PatientLogin')
+    }
+  }, [])
+
+  // Statements for which users must provide initials
+  const agreementStatements = [
+    'I agreed to a 1-week follow-up appointment in the office for a blood pressure check and evaluation followed by monthly appointments for re-eval and script.',
+    'I agree to full disclosure of all my daily medications & medical history that may pertain to Phentermine use.',
+    'I do NOT have Hypertension, Heart Disease, Atherosclerosis, Valvular Heart Disease, Glaucoma, Seizures, Anxiety Disorder or Overactive Thyroid.',
+    'I do NOT have a history of alcohol, drug or substance dependence or abuse.',
+    'I am NOT pregnant or breast feeding.',
+    'I understand Phentermine is intended for short-term use only (1-3 months or as determined between my health care provider and myself.)',
+    'I understand Phentermine may be associated with physical and psychological dependence.',
+    'I will NOT take any other diet medication (over the counter or prescribed) while on Phentermine.',
+    'I agree to return to the office for evaluation of any side effects or problems associated with Phentermine use.',
+    'I will not drive or operate machinery till I know how Phentermine will affect me.',
+    'I understand that lost/stolen prescriptions will not be refilled early, nor will Phentermine be prior authorized if not covered by my insurance plan.',
+    'I will not stop Phentermine suddenly without talking to my health care provider.',
+    'I have received and reviewed PHENTERMINE FACTS sheet.',
+  ]
+
+  // Handles changes in the initials fields
+  const handleInitialChange = (index: number, value: string) => {
+    const newInitials = [...initials]
+    newInitials[index] = value
+    setInitials(newInitials)
+  }
+
+  // Handles form submission
+  const handleSubmit = () => {
+    // Format date of birth for submission
+    const formattedDob = `${dateOfBirth.month}-${dateOfBirth.day}-${dateOfBirth.year}`
```

Comment:
this is probably something that happens a lot, its best to have a date formatter hepper function to use through out the app.

Reply:
Date formatter has been implemented.